app/api/recommend.py

```
@@ -111,14 +117,23 @@ def hybrid_recommend(
     try:
         cached = r.get(cache_key)
         if cached:
+            ttl = r.ttl(cache_key)
+            latency_ms = int((time.perf_counter() - t0) *1000)
+            logger.info(f"[CACHE][HIT] ttl={ttl} latencyMs={latency_ms} key={cache_key_masked}")
             return json.loads(cached)
+        else: 
+            logger.info(f"[CACHE][MISS] key={cache_key_masked}")
     except Exception as e:
-        logger.error(f"Redis 캐시 확인 중 오류: {e}")
+        logger.error(f"[CACHE][ERROR] stage=get endpoint={endpoint} userId={user_id} key={cache_key_masked} err={e}", exc_info=True)
```

Ruff (0.12.2)

Undefined name `endpoint`

(F821)